### ISSUE TITLE
`onChange(of:perform)` not being called for bound `NSAttributedString` when the editor is typed into

### ISSUE THREAD

**Issue by biajoeknee** [danielsaidi/RichTextKit#64]
I'm trying to get an action to occur whenever the text of the editor changes however when I add an `onChange(of:perform)` callback to the editor view it never gets called. The only time it is actually called is if the `@State` field is mutated directly from within the view that defines it.

**Comment by DavidAlvarezDev** [comment]
Not sure if this helps, but it looks like the onChange modifier is called before the bound text is updated. I found this out by troubleshooting an issue I had today with the editor.

What property are you trying to observe? You might look into adding a didSet { } somewhere on the property depending on your setup.

**Comment by bee-uh-joe-knee** [comment]
Yeah, that's always the case with `onChange`. I'm trying to observe the actual text that gets bound to `RichTextEditor`, but whenever the text is mutated — via the editor — the `onChange` callback is never called — despite the value actually mutating.

Tried `didSet`, and it was the same deal — not being called despite the `NSAttributedString` property being mutated by `RichTextKit`.

**Comment by danielsaidi** [comment]
Thanks for reporting @bee-uh-joe-knee - I will take a look at this.

**Comment by asabhaney** [comment]
I might be nuts, but I'm encountering this issue as well. On iOS it seems to work fine, but not on macOS. I've also tried to isolate this issue by creating a new `NSViewRepresentable` from scratch that wraps around `NSTextView`, and the `onChange` doesn't seem to fire there either unless a new `NSAttributedString` instance is assigned to the Binding after each change (which causes other problems).

**Comment by biajoeknee** [comment]
yeah, it's been a while and haven't touched the project that was originally using this, but i too was building for macOS when running into this issue

**Comment by chrisdmacrae** [comment]
This bug sucks; makes RichTextKit unusable for MacOS

**Comment by danielsaidi** [comment]
Hi all! I can confirm that I can also see that the print doesn't work on macOS. It's very strange, because the string does in fact change.

I tried adding this to the demo app, instead of the code above:

```swift
struct EditorScreen: View {

    ....

    var body: some View {
        VStack(spacing: 0) {
            Divider()
            HStack(spacing: 0) {
                editor
                Divider()
                toolbar
            }
        }
        .toolbar {
            ToolbarItem(placement: .automatic) {
                RichTextAction.ButtonStack(
                    context: context,
                    actions: [.undo, .redo, .copy]
                )
            }
        }
        .viewDebug()
        .onChange(of: context.selectedRange) { _ in   <--- HERE
            print(text.string)
        }
    }
}
```

Subscribing to the selected range DOES work, and printing the text correctly prints the current text.

@asabhaney @biajoeknee @chrisdmacrae Can you confirm if this works for you?

**Comment by Kinark** [comment]
I can confirm onChange doesn't detect anything on macOS.

I got to make it work by using `.onReceive` like so:
```
        .onReceive(context.objectWillChange) { _ in
            // Do whatever you wanna do
        }
```
It'll be called much more times than needed, but it works.

**Comment by S1D1T1** [comment]
I do use .onChange with RTK successfully, for syntax coloring. My observer is set up like so:

```
.onChange(of:context.attributedString.string){
...
  }

```
